var/spack/repos/builtin/packages/py-networkx/package.py

```diff
@@ -26,32 +28,43 @@ class PyNetworkx(PythonPackage):
     version("1.11", sha256="0d0e70e10dfb47601cbb3425a00e03e2a2e97477be6f80638fef91d54dd1e4b8")
     version("1.10", sha256="ced4095ab83b7451cec1172183eff419ed32e21397ea4e1971d92a5808ed6fb8")
 
+    # variant would be availabel for earlier versions as well, but then the
+    # dependencies increase a lot
     variant(
         "extra",
-        description="Optional requirements that may require extra steps to install",
         default=False,
+        when="@2.5:",
+        description="Optional requirements that may require extra steps to install",
     )
 
-    depends_on("python@2.7:", type=("build", "run"))
-    depends_on("python@3.5:", when="@2.3:", type=("build", "run"))
-    depends_on("python@3.6:", when="@2.5:", type=("build", "run"))
-    depends_on("python@3.7:", when="@2.6:", type=("build", "run"))
     depends_on("python@3.8:", when="@2.7:", type=("build", "run"))
+    depends_on("python@3.7:", when="@2.6:", type=("build", "run"))
     depends_on("py-setuptools", type="build")
-    depends_on("py-decorator@3.4.0:", when="@:1", type=("build", "run"))
-    depends_on("py-decorator@4.1.0:", when="@2.0:2.1", type=("build", "run"))
-    depends_on("py-decorator@4.3.0:", when="@2.2:2.4", type=("build", "run"))
-    depends_on("py-decorator@4.3.0:4", when="@2.5.1:2.5", type=("build", "run"))
+
     # From requirements/default.txt
```

Comment:
Curious what you think about this. By default, `install_requires` is empty, so `pip install networkx` doesn't install anything. You have to use `pip install networkx[default]` to get these. Should we add a variant (default to True) to control these, or leave it as is?

Reply:
Alright, let's add a variant

Reply:
Done